### PR TITLE
Tutorial refactor

### PR DESCRIPTION
**Pull request by BorisJov**

### CODE REVIEW

File: Assets/MirageXR/Tutorial/HelpSteps/HelpSelectionModel.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    /// <summary>
+    /// Model for all items in the help selection dialogue.
+    /// Should be serialisable.
+    /// </summary>
+    public class HelpSelectionModel
+    {
+        /// <summary>
+        /// Text that should be shown in the help popup, when the user is selecting help.
+        /// </summary>
+        [JsonProperty("selection_text")] public string SelectionText { get; set; }
```
Comment (cawke) [inline Assets/MirageXR/Tutorial/HelpSteps/HelpSelectionModel.cs]:
```suggestion
        [JsonProperty] public string SelectionText { get; set; }
```

for others too

File: Assets/MirageXR/Player/Scripts/Mobile/Popup/HelpSelectionPopup.cs [hunk 2]
```diff
@@ -41,5 +143,28 @@ protected override bool TryToGetArguments(params object[] args)
...
+        /// </summary>
+        /// <param name="steps">The steps in the mini-tutorial.</param>
+        private void ShowShortMessageSequence(List<TutorialModelUI> steps)
+        {
+            this.Close();
+            var queue = new Queue<TutorialModelUI>(steps);
+            TutorialManager.Instance.MobileTutorial.Show(queue);
+        }
+
+        private void startMobileEditingTutorial()
```
Comment (cawke) [inline Assets/MirageXR/Player/Scripts/Mobile/Popup/HelpSelectionPopup.cs]:
```suggestion
        private void StartMobileEditingTutorial()
```

File: Assets/MirageXR/Player/Scripts/Mobile/Popup/HelpSelectionPopup.cs [hunk 3]
```diff
@@ -41,5 +143,28 @@ protected override bool TryToGetArguments(params object[] args)
...
+        {
+            this.Close();
+            TutorialManager.Instance.StartNewMobileEditingTutorial();
+        }
+
+        private void startMobileViewingTutorial()
```
Comment (cawke) [inline Assets/MirageXR/Player/Scripts/Mobile/Popup/HelpSelectionPopup.cs]:
```suggestion
        private void StartMobileViewingTutorial()
```

**Review by cawke — CHANGES_REQUESTED** [review]
generally good, correct minor errors and check the positions of all windows so that the situation does not occur as in the screenshot

image

File: Assets/MirageXR/Tutorial/HelpSteps/HelpSelectionModel.cs (hunk 1 above)
Comment (BorisJov) [reply Assets/MirageXR/Tutorial/HelpSteps/HelpSelectionModel.cs]: Why remove these? How will it parse then?

Comment (BorisJov) [reply Assets/MirageXR/Tutorial/HelpSteps/HelpSelectionModel.cs]: I've just tried removing the names and it doesn't work without them.

**Pull request merged**